### PR TITLE
carrot shivs strike back

### PR DESCRIPTION
**Pull request by PKPenguin321**
rerun of https://github.com/tgstation/-tg-station/pull/14857 because git raped me

Carrot shivs, made by applying any kind of knife/a botany hatchet to a carrot. They deal 8 damage on hit, 12 on throw.

:cl: PKPenguin321
rscadd: You can now sharpen carrots into shivs by using a knife or hatchet on them.
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
>196 lines removed

okay what the fuck
i guess git is going back in for round two
god dammit

**Comment by WJohn** [comment]
-196 lines of just empty space.
:+1: on that alone.

**Comment by PKPenguin321** [comment]
what i meant was yes i totally meant to remove the empty spaces

**Comment by KorPhaeron** [comment]
@WJohn 

Removing the empty spaces between different objects isn't that good

**Comment by PKPenguin321** [comment]
what i meant was i have no idea how that happened git it playing mindgames with me

**Comment by freerealestate** [comment]
this is dumb

**Comment by PKPenguin321** [comment]
you're dumb fuck off

**Comment by freerealestate** [comment]
be well :)

**Comment by PKPenguin321** [comment]
im gonna be completely honest, i dont know how the fucking line changes happened and im just so fucking done with git right now

if this is deemed unacceptable by a maintainer i guess i'll try to fix it but im really kinda burned out right now

**Comment by tkdrg** [comment]
Please don't remove these blank lines
Also you kinda forgot to add a way to actually make those shivs

**Comment by Chocobro** [comment]
:+1: however i'm concerned it's too much damage
it is a fucking carrot after all

**Comment by PKPenguin321** [comment]
it's less damage than a metal rod

**Comment by PKPenguin321** [comment]
>Also you kinda forgot to add a way to actually make those shivs

no, it's just hidden in all the removed lines. line 888 in grown.dm
ill try to fix the randomly removed lines

**Comment by Iamgoofball** [comment]
>ill try to fix the randomly removed lines

Ignore them, they're whitespace changes. Doesn't matter.

**Comment by dannno** [comment]
>it's less damage than a metal rod

yeah but its still a carrot lmao

**Comment by Chocobro** [comment]
hit yourself in the head with a carrot
then hit yourself in the head with a fucking metal pole
the carrot really won't do a lot other than maybe snap

**Comment by PKPenguin321** [comment]
it's a very sharp carrot

**Comment by Chocobro** [comment]
:ok_hand:

**Comment by dannno** [comment]
it's gonna be hella easy to kill someone with carrots with that throw force

**Comment by Chocobro** [comment]
okay time to turn off email notifications for git

**Comment by Chocobro** [comment]
yep there's the email which contains my reply that's real fucking good to know

**Comment by PKPenguin321** [comment]
>requires nine consecutive successful throws ONLY IF the target has 0 armor, if they have any armor it takes as many as floor tiles
>less portable than floor tiles (floor tiles take like one spot in your backpack and stack to fifty and are found pretty much anywhere, carrots need a slot for each individual one)

carrots are just funnier and function as a knife
can't you people just appreciate something

**Comment by freerealestate** [comment]
not carrot

**Comment by PKPenguin321** [comment]
line removal problem fixed

File: code/game/objects/items/weapons/kitchen.dm [hunk 1]
```diff
@@ -98,6 +98,14 @@
 	origin_tech = "materials=2;combat=4"
 	attack_verb = list("slashed", "stabbed", "sliced", "torn", "ripped", "cut")
 
+/obj/item/weapon/kitchen/knife/carrotshiv
```
Comment (phil235) [inline code/game/objects/items/weapons/kitchen.dm]: Do we really want to be able to butcher a xenomorph with a pointy carrot?

Comment (PKPenguin321) [inline code/game/objects/items/weapons/kitchen.dm]: yes.

**Comment by AlexanderUlanH** [comment]
>Fun-hating realism police are opposed to the idea

That's how you know you're doing something right.  :+1: because we should be able to kill people by stuffing a backpack full of sharpened carrots and throwing them.

**Comment by PKPenguin321** [comment]
thanks fam

**Comment by freerealestate** [comment]
>carrots
>fun

bugs?

**Comment by PKPenguin321** [comment]
>being so jaded that you derive no enjoyment and cannot imagine others deriving enjoyment from the absurdity of knife-carrots

why do you even play video games if you're such an emotionless fun-hating android tbh

**Comment by freerealestate** [comment]
this argument is pretty absurd and sharpened carrots are dumb
not much else to add really

**Comment by TrustyGun** [comment]
you missed an opportunity with naming the title "carrot shivs shiv back"

Otherwise: :+1:

File: code/game/objects/items/weapons/kitchen.dm (hunk 1 above)
Comment (KazeEspada) [inline code/game/objects/items/weapons/kitchen.dm]: However, if you make them a child of screwdriver, you could stab peoples eyes out and unscrew windows with it.

**Comment by PKPenguin321** [comment]
This has been ready to merge since it was opened. Merge-Ready tag or merge, please. @tkdrg @remierichards @duncathan @dorsidwarf @razharas

File: code/modules/hydroponics/grown.dm [hunk 2]
```diff
@@ -1034,6 +1034,15 @@ obj/item/weapon/reagent_containers/food/snacks/grown/shell/eggy/add_juice()
 		reagents.add_reagent("oculine", 3+round(potency / 5, 1))
 		bitesize = 1 + round(reagents.total_volume / 2, 1)
 
+/obj/item/weapon/reagent_containers/food/snacks/grown/carrot/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/kitchen/knife) || istype(I, /obj/item/weapon/hatchet))
```
Comment (duncathan) [inline code/modules/hydroponics/grown.dm]: >can sharpen a carrot using another carrot
:+1:

File: code/game/objects/items/weapons/kitchen.dm (hunk 1 above)
Comment (duncathan) [inline code/game/objects/items/weapons/kitchen.dm]: The people asked for carrot shivs, not shanks

Comment (duncathan) [inline code/game/objects/items/weapons/kitchen.dm]: Anyway this sounds fun to me

**Comment by Chocobro** [comment]
i can't wait for the feedback topic full of salty players

**Comment by duncathan** [comment]
I seriously can't believe that some people think I hate fun

File: code/modules/hydroponics/grown.dm (hunk 2 above)
Comment (PKPenguin321) [inline code/modules/hydroponics/grown.dm]: they are very sharp carrots.

**Comment by PKPenguin321** [comment]
Remember when people said this would be op

**Comment by jughu** [comment]
I can fit 6 glass shards a cannon but i only can fit 3 carrot shivs inside?... what is this magic

**Comment by Iamgoofball** [comment]
needs to use is_sharp()

**Pull request merged**